Nodejs/Product/TestAdapter/TestFrameworks/Tape/tape.js

```diff
@@ -57,45 +51,55 @@ function run_tests(testInfo, callback) {
         return;
     }
 
+    // Since the test events don't come in order we store all of them in this array
+    // in the 'onFinish' event we loop through them and process anything remaining.
+    var testState = [];
     var harness = tape.getHarness({ objectMode: true });
-    var capture = false; // Only capture between 'test' and 'end' events to avoid skipped test events.
+
     harness.createStream({ objectMode: true }).on('data', function (evt) {
         switch (evt.type) {
             case 'test':
-                capture = true;
-                // Test is starting. Reset the result object. Send a "test start" event.
-                result = {
+
+                var result = {
                     'title': evt.name,
-                    'passed': true,
+                    'passed': undefined,
                     'stdOut': '',
                     'stdErr': ''
                 };
+
+                testState[evt.id] = result;
+
+                // Test is starting. Reset the result object. Send a "test start" event.
                 callback({
                     'type': 'test start',
                     'title': result.title,
                     'result': result
                 });
                 break;
             case 'assert':
-                if (!capture) break;
+                var result = testState[evt.test];
```

Comment:
I think this case is incorrect. Based on the Tape documentation a test can contain multiple assertions and tape well report the result for each one of them.
This should be change to either store all of the assertions and later check its result during the 'end' case; or assume the test is successful until any failed assertion proves it otherwise (which the previous revision was doing).